generic/multidispatch.py

```diff
@@ -126,7 +127,7 @@ def __call__(self, *args: Any, **kwargs: Any) -> Any:
         trimmed_args = args[: self.params_arity]
         rule = self.registry.lookup(*trimmed_args)
         if not rule:
-            print(self.registry._tree)
+            logging.error(self.registry._tree)
```

Comment:
I think it should be a `debug` message instead, since it shows a dump of the registry tree.

Also it would be better to define a separate logger on top (`log = logging.getLogger(__name__)`). `logging.* uses the root logger.